main_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIndexRoute(t *testing.T) {
+	// Define a structure for specifying input and output
+	// data of a single test case. This structure is then used
+	// to create a so called test map, which contains all test
+	// cases, that should be run for testing this function
+	tests := []struct {
+		description string
+
+		// Test input
+		route string
+
+		// Expected output
+		expectedError bool
+		expectedCode  int
+		expectedBody  string
+		method        string
+	}{
+		{
+			description:   "publishers get route",
+			route:         "/publishers",
+			method:        "GET",
+			expectedError: false,
+			expectedCode:  200,
+			expectedBody:  "[]",
+		},
+		{
+			description:   "non existing route",
+			route:         "/i-dont-exist",
+			method:        "GET",
+			expectedError: false,
+			expectedCode:  404,
+			expectedBody:  "{\"message\":\"Cannot GET /i-dont-exist\"}",
+		},
+	}
+
+	os.Setenv("DATABASE_DSN", "host=localhost user=postgres password=postgres dbname=postgres port=5432")
```

Comment:
I think we should either have a clean test db or mock it, because otherwise we'd clutter the development db when we test for the endpoints that write resources.

Reply:
Yes, that's right. I've put here in order to change in the future, we could consider a test db for local environment.